lenskit-core/src/main/java/org/grouplens/lenskit/data/dao/packed/BinaryRatingPacker.java

```diff
@@ -252,7 +251,6 @@ private void upgradeRatings(BinaryFormat newFormat) throws IOException {
 
         ByteBuffer oldBuffer = ByteBuffer.allocateDirect(format.getRatingSize());
         ByteBuffer newBuffer = ByteBuffer.allocateDirect(newFormat.getRatingSize());
-        MutableRating scratch = new MutableRating();
```

Comment:
Good :). I'm wanting to get rid of a lot of that garbage, and re-introduce carefully controlled and better-designed versions of it where performance measurement shows that it is needed.